Adds a regular expression for device naming for #50 issue

TheSkorm opened the pull request:
This adds validation (like attributes) for hostnames which should resolve issue #50

jathanism commented:
Hi there. Thanks for the patch! Can you please make the pattern inclusive to allow for valid hostname characters? As you can see, your patch broke the unit tests that are using hostnames like `foo-bar1` (where a hyphen is a valid hostname character).

If you could fix the pattern, and also implement a couple minor unit tests for it exercising the hostname validation, that would be excellent! Thanks!

TheSkorm commented:
Will do. Didn't see the unit tests, will have a look

On Sat, 20 Feb 2016 09:38 Jathan McCollum <notifications@github.com> wrote:

> Hi there. Thanks for the patch! Can you please make the pattern inclusive
> to allow for valid hostname characters? As you can see, your patch broke
> the unit tests that are using hostnames like foo-bar1 (where a hyphen is
> a valid hostname character).
>
> If you could fix the pattern, and also implement a couple minor unit tests
> for it exercising the hostname validation, that would be excellent! Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dropbox/nsot/pull/149#issuecomment-186436971>.
>

TheSkorm commented:
Regexp has been updated and added unit tests.

jathanism commented:
Awesome! So, last steps:

- Increment the version in `nsot/version.py` to `0.14.1`
- Rebase everything to a single commit!

Thanks for your contribution!

jathanism commented on tests/api_tests/test_devices.py:
```diff
@@ -34,6 +34,15 @@ def test_creation(live_server, user, site):
 
     admin_client.create(attr_uri, resource_name='Device', name='attr1')
 
+    # Test invalid attribute name
```
*hostname (not attribute name)

TheSkorm commented:
Rebased and fixed up that comment (I wonder where I borrowed that code from ;) ). Version number also updated.

Thanks for guiding me through the pull process. Looking forward to working on some other bugs.

jathanism commented:
@TheSkorm My pleasure. And thanks again for your contribution. Please consider joining us at #nsot on IRC (Freenode).
